tests/test_string_based_date_time.py

```diff
@@ -0,0 +1,74 @@
+from datetime import datetime, timedelta
+
+import arrow
+import pytest
+
+from implicitdict import StringBasedDateTime
+
+
+ZERO_SKEW = timedelta(microseconds=0)
+
+
+def test_consistency():
+    t = datetime.utcnow()
+    sbdt = StringBasedDateTime(t)
+    with pytest.raises(TypeError):  # can't subtract offset-naive and offset-aware datetimes
+        assert abs(sbdt.datetime - t) <= ZERO_SKEW
+    with pytest.raises(TypeError):  # can't compare offset-naive and offset-aware datetimes
+        assert not (sbdt.datetime > t)
+    assert abs(sbdt.datetime - arrow.get(t).datetime) <= ZERO_SKEW
+
+    sbdt = StringBasedDateTime(t.isoformat())
+    assert abs(sbdt.datetime - arrow.get(t).datetime) <= ZERO_SKEW
+
+    t = arrow.utcnow().datetime
+    sbdt = StringBasedDateTime(t)
+    assert abs(sbdt.datetime - t) <= ZERO_SKEW
+
+    sbdt = StringBasedDateTime(t.isoformat())
+    assert abs(sbdt.datetime - t) <= ZERO_SKEW
+
+    t = arrow.now('US/Pacific').datetime
+    sbdt = StringBasedDateTime(t)
+    assert abs(sbdt.datetime - t) <= ZERO_SKEW
+
+    sbdt = StringBasedDateTime(t.isoformat())
+    assert abs(sbdt.datetime - t) <= ZERO_SKEW
+
+    t = arrow.now('US/Pacific')
+    sbdt = StringBasedDateTime(t)
+    assert abs(sbdt.datetime - t) <= ZERO_SKEW
+
+    sbdt = StringBasedDateTime(t.isoformat())
+    assert abs(sbdt.datetime - t) <= ZERO_SKEW
+
+    t = arrow.get("1234-01-23T12:34:56.12345678")
+    sbdt = StringBasedDateTime(t)
+    assert "12345678" not in sbdt  # arrow only stores integer microseconds
+
+    with pytest.raises(ValueError):  # unconverted data remains (datetime only parses down to microseconds)
+        _ = datetime.strptime("1234-01-23T12:34:56.12345678", "%Y-%m-%dT%H:%M:%S.%f")
```

Comment:
Out of curiosity, is `_ = ` required ?

Reply:
No, not required.  I was initially trying to write things that people might attempt and that would usually involve assignment, but assignment isn't part of the mechanism that demonstrates the behavior.  Removed.